src/EC2Store.js

```diff
@@ -22,7 +22,59 @@ class EC2Store {
 			let ec2 = new AWS.EC2();
 			ec2.describeSnapshots({}, function (err, response) {
 				if (err) reject(err);
-				else resolve(response);
+
+				else {
+					var taggedForBackup = response.Snapshots.filter(function(obj){
+						for (var i=0; i<obj.Tags.length; i++) {
+							if (obj.Tags[i].Key === 'backups:config-v0') {
+								return true;
+							}
+						}
+						return false;
+					});
+
+					resolve(taggedForBackup.map(function(obj){
+
+						var filteredForName = obj.Tags.filter(function(obj){
+							if (obj.Key === 'Name'){
+								return true;
+							} else {
+								return false;
+							}
+						});
+
+						if (filteredForName.length === 1) {
+							filteredForName = filteredForName[0];
+						} else {
+							throw new Error('expected to receive snapshot with a single value for name but length > 1');
+						}
+
+						var filteredForDate = obj.Tags.filter(function(obj){
+							if (obj.Value.slice(0,10) === 'ExpiryDate'){
+								return true;
+							} else {
+								return false;
+							}
+						});
+
+						if (filteredForDate.length === 1) {
+							filteredForDate = parseInt(filteredForDate[0].Value.slice(11,23));
```

Comment:
Just a heads up: this line (and the other `slice` line above) will work for now, but in the value of this tag could look different in the future. Something like
```
OtherMetadata:some_random_junk,ExpiryDate:201601011125
```
or
```
ExpiryDate:201601011125,MoreOptions:AString!
```

Can you think of ways that you could isolate just the `ExpiryDate` parts of strings like these?